graftm/tree_decorator.py

```diff
@@ -0,0 +1,304 @@
+################################################################################
+################################# - Imports - ##################################
+
+# Local imports
+from graftm.getaxnseq import Getaxnseq
+
+# System imports
+from skbio import TreeNode
+import argparse
+import logging
+import os
+import sys
+
+################################################################################
+################################ - Statics - ###################################
+MAX_RESOLUTION = 7
+
+################################################################################
+################################## - Code - ####################################
+
+class TreeUnrootedException(Exception): 
+    pass
+
+class TreeDecorator:
+    '''Yet another class that decorates trees, only the way I want it.'''
+    
+    
+    def __init__(self, 
+                 tree,
+                 taxonomy,
+                 seqinfo=None):
+        '''
+        Set up empty lists and open and import tree object, and load the
+        taxonomy into a hash
+        '''
+        
+        self.gg_prefixes = ["k__", 'p__', 'c__', 'o__', 'f__', 'g__', 's__']
+        
+        # Set empty list to record nodes that have already been assigned
+        # taxonomy.
+        self.encountered_nodes = {}
+        self.encountered_taxonomies = set()
+        
+        # Read in tree
+        self.tree = tree
+        
+        # Read in taxonomy
+        if seqinfo:
+            gtns = Getaxnseq()
+            logging.info("Importing taxtastic taxonomy from files: %s and %s" % (taxonomy, seqinfo))
+            self.taxonomy =  gtns.read_taxtastic_taxonomy_and_seqinfo(open(taxonomy), open(seqinfo))
+            for id, taxonomy_list in self.taxonomy.iteritems():
+                if len(taxonomy_list) != 7:
```

Comment:
I see a 7. Why use 7 instead of discovering the number of ranks from the taxonomy?